android_vs_ios.md

```diff
@@ -0,0 +1,175 @@
+iOS developers generally write all mocks manually, partly due to lack of a good mocking framework. There is a library called [Cuckoo](https://github.com/Brightify/Cuckoo) which seems to be slightly more limited version of Mockito.
+
+# Other stuff
+## Equivalent terminologies/classes/frameworks
+|Android/Kotlin | iOS/Swift|
+|---|---|
+|`val` | `let`|
+|`fun` | `func`|
+|`when` | `switch`|
+|`null` | `nil`|
+|`dp` | point (resolution)|
+|Constructor | Init|
+|Interface | Protocol|
+|Lambda | Closure|
+|Kotlin sealed class | Enum|
+|Activity | ViewController|
+|layout xml | xib|
+|ConstraintLayout | Auto Layout|
+|LinearLayout | UIStackView (but more limited)|
+|TextView | UILabel|
+|EditText | UITextField / UITextView|
+|ListView (deprecated, use RecyclerView) | UITableView|
+|RecyclerView | UICollectionView|
+|RxJava | RxSwift|
+|RxBinding | RxCocoa|
+|LiveData | Driver/BehaviorRelay|
+|Retrofit | Alamofire (simple network calls can be done easily without a library)|
+|Gson/Jackson/Moshi | Codable|
+|Picasso/Glide | Kingfisher|
+|Timber | CocoaLumberjack|
+
+## Miscellaneous tidbits
+Since Android is open source, sometimes it's very helpful to look into the source code of framework classes when debugging issues.
+
+AppCode (as of version 2018.2) is not much better than XCode, it’s definitely nowhere near the quality of other JetBrains IDEs like IntelliJ or Android Studio. It’s noticeably slower, can't find code usage sometimes, has issues autocompleting certain extensions, and generally feels unreliable when refactoring.
+
+On Android, Button is just a TextView with a different default styling (same with ImageView/ImageButton). On iOS, UIButton and UILabel are completely different classes. Image button on iOS is just an UIButton with image set instead of text.
+
+On iOS, there are two different views for entering text: UITextField (for single line, and doesn’t support multiline) and UITextView (for multiline, but doesn’t support placeholder). On Android, there’s just EditText with supports both multiline and hints.
+
+Swift has a lot of breaking changes in between major versions. Whereas nothing is ever really removed in Java.
+
+iOS 11 finally added ability to use named colors (similar to colors.xml on Android) in the interface builder. Before that, you need to specify the color RGB manually every time…
```

Comment:
Semi-related to this, is there anything worth saying about how the platforms handle string localization? Or other resource types like images?

Reply:
yeah, it would be a good idea to add a section on localization and resource management. However, I don't have enough experience on this on the iOS side so someone else with better knowledge should create one